packages/core/index.js

```diff
@@ -1,78 +1,4 @@
- * @return {middy} - a `middy` instance
- */
-
-/**
- * @typedef pluginObject
- * @type Object
- * @property {function} beforePrefetch - request hook for *beforePrefetch*
- * @property {function} requestStart - request hook for *beforePrefetch*
- * @property {function} beforeMiddleware - request hook for *beforeMiddleware*
- * @property {function} afterMiddleware - request hook for *afterMiddleware*
- * @property {function} beforeHandler - request hook for *beforeHandler*
- * @property {function} afterHandler - request hook for *afterHandler*
- * @property {function} requestEnd - request hook for *beforePrefetch*
- */
-
-/**
- * @typedef middy
- * @type function
- * @param {Object} event - the AWS Lambda event from the original handler
- * @param {Object} context - the AWS Lambda context from the original handler
- * @property {useFunction} use - attach one or more new middlewares
- * @property {applyMiddlewareFunction} applyMiddleware - attach a new middleware
- * @property {middlewareAttachFunction} before - attach a new *before-only* middleware
- * @property {middlewareAttachFunction} after - attach a new *after-only* middleware
- * @property {middlewareAttachFunction} onError - attach a new *error-handler-only* middleware
- * @property {Object} __middlewares - contains the list of all the attached
- *   middlewares organised by type (`before`, `after`, `onError`). To be used only
- *   for testing and debugging purposes
- */
-
-/**
- * @typedef useFunction
- * @type {function}
- * @param {middlewareObject|middlewareObject[]} - the middleware object or array of middleware objects to attach
- * @return {middy}
- */
-
-/**
- * @typedef middlewareObject
- * @type Object
- * @property {middlewareFunction} before - the middleware function to attach as *before* middleware
- * @property {middlewareFunction} after - the middleware function to attach as *after* middleware
- * @property {middlewareFunction} onError - the middleware function to attach as *error* middleware
- */
-
-/**
- * @typedef applyMiddlewareFunction
- * @type {function}
- * @param {middlewareObject} - the middleware object to attach
- * @return {middy}
- */
-
-/**
- * @typedef middlewareAttachFunction
- * @type {function}
- * @param {middlewareFunction} - the middleware function to attach
- * @return {middy}
- */
-
-/**
- * @typedef middlewareFunction
- * @type {function}
- * @param {function} handler - the original handler function.
- *   It will expose properties `event`, `context`, `response`, `error` and `internal` that can
- *   be used to interact with the middleware lifecycle
- * @return {void|Promise} - A middleware can return a Promise.
- *                          In this case middy will wait for the promise to resolve (or reject) and it will automatically
- *                          propagate the result to the next middleware.
- */
-
-const middy = (handler = () => {}, plugin) => {
+const middy = (handler = () => { }, plugin = null) => {
```

Comment:
any reason you're casting plugin to `null`? undefined is considered nullish

Reply:
I was trying to make it optional... Maybe i should revert that!